docs/overview/04-proof-aggregation/03-publish-attestations.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Publish Attestations
+---
+
+Every time an aggregation is complete a [`AggregationComplete{domainId, aggregationId}`](../02-mainchain/05-mainchain_api.md#aggregationcomplete) event is emitted; the aggregation now is complete, no more proofs can be added to it and is moved in the publishing queue.
+
+In order to generate an aggregation receipt (the Merkle root) it's sufficient that any user calls [`aggregate(domainId, aggregationId)`](../02-mainchain/05-mainchain_api.md#aggregate_3) with the correct coordinates. Build this aggregation has a cost because you should pay the effort to generate the tree, compute the root and emit the events, but the user will receive the money that was collected by the [proof submitters](./02-handle-valid-proof.md#proof-submission-costs). The publisher should take care that the estimations did to hold the money from the proof submitters can be different from the actual price: in the normal flow a publisher should evaluate if it's worth of generate the aggregation by inspect the storage before doing the call.
+
+If the `aggregate` succeeds a [`NewAggregationReceipt{domainId, aggregationId, receipt}`](../02-mainchain/05-mainchain_api.md#newaggregationreceipt) event is emitted in this block. Now the proof submitters can use the [`aggregate_statementPath`](../02-mainchain/05-mainchain_api.md#aggregate_statementpath) RPC to compute
+the path of the statement in this receipt: don't forget to save the block hash where the event is emitted to use it in the RPC call.
+
+## Publish an Incomplete Aggregation
+
+It's also possible to call `aggregate` extrinsic on an incomplete aggregation: in this case the publisher will receive just the money put on hold till this moment. This can be useful when:
+
+- a submitter would generate his aggregation without waiting it become complete and pay himself the difference
+- a publisher think that it is already convenient for him to proceed with the aggregation because the accumulated funds just cover the execution price
```

Comment:
```suggestion
- a publisher thinks that it is already convenient for him to proceed with the aggregation because the accumulated funds are enough to cover for the execution cost
```